django/db/backends/mysql/introspection.py

```diff
@@ -147,6 +147,16 @@ def get_storage_engine(self, cursor, table_name):
             return self.connection.features._mysql_storage_engine
         return result[0]
 
+    def _parse_constraint_columns(self, sql, columns):
+        cols = OrderedSet()
+        statement = sqlparse.parse(sql)[0]
+        tokens = (token for token in statement.flatten() if not token.is_whitespace)
+        for token in tokens:
+            if token.ttype == sqlparse.tokens.Name:
+                if token.value[1:-1] in columns:
+                    cols.add(token.value[1:-1])
```

Comment:
We assume (and it seems that it is true) that columns names are always quoted, we should use this here to fix cases when columns names overlap with function names (see tests with `LOWER`), e.g.
```python
if (
    token.ttype == sqlparse.tokens.Name and
    self.connection.ops.quote_name(token.value) == token_value and
    token.value[1:-1] in columns
):
    cols.add(token.value[1:-1])
```

Reply:
Done.